tomplot/data_manipulation.py

```diff
@@ -0,0 +1,42 @@
+import pandas as pd
+import numpy as np
+
+
+def area_restriction(field_data, coords_X, coords_Y, coord_lims):
+    """
+    A function that restricts a data field and returns the new restricted field
+    as well as its corresponding co-ordinates.
+
+    Args:
+        field_data (':class:`numpy.ndarray`): The field to be filtered.
+        coords_X (':class:`numpy.ndarray`): The field along the X axis.
+        coords_Y (':class:`numpy.ndarray`): The field along the Y axis.
+        coord_lims (':class:'python dictionary): A dicitionary containing the
+            coordinates as Keys and the values are tuples containing the limits.
+
+    Returns:
+        new_field_data (':class:`numpy.ndarray`): The new restricted data.
+        new_X_coords (':class:`numpy.ndarray`): The X coordinates for the restricted data
+        new_Y_coords (':class:`numpy.ndarray`): The Y coordinates for the restricted data
+    """
+
+    if len(np.shape(field_data)) != 1:
+        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
+    if len(np.shape(coords_X)) != 1:
+        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
+    if len(np.shape(coords_Y)) != 1:
+        raise ValueError('area_restriction: input data must be 1D to be filtered by pandas data frame')
+
```

Comment:
Sounds sensible! Should we also include a check for dictionaries having more than two keys? I think if we want to evaluate multiple fields at once, it may perhaps make more sense to just have multiple calls of the restrict area function instead of encoding that functionality into the routine itself